src/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -196,8 +197,11 @@ protected void purchase(final boolean purchaseForBid, int PUsToSpend, final IPur
 
       // Check if any place territories exist
       final Map<Territory, ProPurchaseTerritory> purchaseTerritories = purchaseUtils.findPurchaseTerritories(player);
-      if (purchaseTerritories.isEmpty()) {
-        ProLogger.info("No possible place territories owned so exiting purchase logic");
+      final List<Territory> possibleFactoryTerritories =
```

Comment:
PS: fix as necessary otherwise means make it so it's not confusing that you're looking for "possible territories", which to means territories you could place on, yet you are looking for "no infra" and not captured, when it sounds like you should be looking for infrastructure (factories). Hence the confusion I have after reviewing the code.  Given the invert I'm not going to update the name of hte method, that would be incorrect.